R/zchunk_LA119.solar.R

```diff
@@ -29,10 +29,37 @@ module_gcam.usa_LA119.Solar_DISABLED <- function(command, ...) {
     # Load required inputs
     states_subregions <- get_data(all_data, "gcam-usa/states_subregions")
     NREL_us_re_capacity_factors <- get_data(all_data, "gcam-usa/NREL_us_re_capacity_factors")
-    NREL_us_re_technical_potential <- get_data(all_data, "gcam-usa/NREL_us_re_technical_potential")
+    NREL_us_re_technical_potential <- get_data(all_data, "gcam-usa/NREL_us_re_technical_potential")   #This data is not being used in this chunk
+
+    # Perform computations: Create scalars to scale capacity factors read in in the assumptions file in the energy folder to adjust for varying irradiance by state.
+
+    NREL_us_re_capacity_factors %>% gather(fuel,average,-State) %>%
+      filter (State=="Average") %>%
+      select (-State) -> Average_fuel
+
+    NREL_us_re_capacity_factors %>%
+      gather(fuel,value,-State) %>%
+      filter (State != "Average") %>%
+      full_join(Average_fuel, by = "fuel") %>%
```

Comment:
I see a couple of problems with the data and above code. The original data has an average row, representing the average of the capacity factors, not including the 0 capacity factors. Calculating the average is what we should be using the DSR for, rather using a read in value. If someone changes the capacity factor for a state, the average would be wrong. I also noticed the average is slightly off. Please rewrite to do in code.

You should gather only once and then separate out the average from the state data. Then you can take the average of the state data, or use the read-in average for old code behavior. The left_join_error_no_match should be used. We want to ensure values for all state.